app/controllers/meals_controller.rb

```diff
@@ -0,0 +1,10 @@
+class MealsController < ApplicationController
+  include MealsHelper
+  def create
+    @meal = Meal.new(meal_params)
+    @meal.menu_id = params[:menu_id]
```

Comment:
You can create private method with memoization:
```
def menu
   @menu ||= Menu.find_by(id: params[:menu_id])
end
```
and initialize associations next way
```
@meal = menu.meals.new(meal_params)
```
This line should be removed